src/render/image_atlas.js

```diff
@@ -123,6 +123,7 @@ export default class ImageAtlas {
     }
 
     patchUpdatedImages(imageManager: ImageManager, texture: Texture) {
+        this.haveRenderCallbacks = this.haveRenderCallbacks.filter(id => imageManager.hasImage(id));
```

Comment:
Instead of filtering, could `dispatchRenderCallbacks` check that the image still exists before proceeding?

Reply:
We can do that, but the image reference would persist in `haveRenderCallbacks` in `ImageAtlas`, and we continue to call `dispatchRenderCallbacks`.

We can mutate the `haveRenderCallbacks` reference in `dispatchRenderCallbacks`, but I think that it may lead to race conditions.

```js
imageManager.dispatchRenderCallbacks(this.haveRenderCallbacks);
```